File: playground_series/s3e8/jl/baseline.jl
using CSV
using Statistics
using DataFrames

trn = CSV.read("./playground_series/s3e8/data/train.csv", DataFrame)
tst = CSV.read("./playground_series/s3e8/data/test.csv", DataFrame)

#so this is the diamonds dataset

pred = mean(trn.:price)

sub = DataFrame(
    id = tst.:id,
    price = pred
)

CSV.write("./playground_series/s3e8/submissions/baseline.csv", sub)

File: playground_series/s5e1/baseline.py
import polars as pl
import numpy as np
import statistics

trn = pl.read_csv("playground_series/s5e1/data/train.csv")
tst = pl.read_csv("playground_series/s5e1/data/test.csv")

#let's just take the rounded mean for now
#although I should probably come up with a better way to handle the nulls
y = trn['num_sold'].drop_nulls()

avg = round(statistics.mean(y))

preds = pl.DataFrame(
    {
        "id": tst["id"],
        "num_sold": avg
    }
)

preds.write_csv("playground_series/s5e1/submissions/baseline.csv")